browse-at-remote.el

```diff
@@ -253,6 +255,47 @@ If HEAD is detached, return nil."
   "Commit URL formatted for github"
   (format "%s/commit/%s" repo-url commithash))
 
+(defun browse-at-remote-ado-format-url (repo-url)
+  "Get an ado formatted URL."
+  (let* ((s (split-string repo-url "/")))
+    ;; [protocol]//[organization].visualstudio.com/[project]/_git/[repository]
+    (format "%s//%s/%s/_git/%s"
+            (nth 0 s)
+            (replace-regexp-in-string "^vs-ssh" (nth 4 s) (nth 2 s))
+            (nth 5 s)
+            (nth 6 s)
+            )
+    )
+  )
+
+(defun browse-at-remote--format-region-url-as-ado (repo-url location filename &optional linestart lineend)
+  "URL formatted for ado"
+  (let* (
+         ;; NOTE: I'm not sure what's the meaning of the
+         ;; branch-prefix. My guess is that it stands for a "Git
+         ;; Branch".
+         (branch-prefix "GB")
+         (base-url (format "%s?version=%s%s&path=/%s"
+                           (browse-at-remote-ado-format-url repo-url)
+                           branch-prefix
+                           location
+                           filename))
+         )
+  (cond
+   ((and linestart lineend)
+    (format "%s&line=%d&lineEnd=%d&lineStartColumn=1&lineEndColumn=1" base-url linestart (+ 1 lineend)))
+   (linestart (format "%s&line=%d&lineStartColumn=1&lineEndColumn=1" base-url linestart))
+   (t base-url))
```

Comment:
pls put closing parentheses on same line, refer https://github.com/bbatsov/emacs-lisp-style-guide

Reply:
Done, though I hate that convention. It makes hard to move the last line up, or copy it and paste in another place.